mindee/parsing/custom/generated_object.py

```diff
@@ -0,0 +1,70 @@
+from typing import Optional
+
+from mindee.parsing.common.string_dict import StringDict
+
+
+class GeneratedObjectField:
+    """A JSON-like object, with miscellaneous values."""
+
+    page_id: Optional[int]
+    """Id of the page the object was found on"""
+
+    def __init__(
+        self,
+        raw_prediction: StringDict,
+        page_id: Optional[int] = None,
+    ) -> None:
+        item_page_id = None
+        for name, value in raw_prediction.items():
+            if name == "page_id":
+                item_page_id = value
+            elif name in ["polygon", "rectangle", "quadrangle", "bounding_box"]:
+                self.__setattr__(
+                    name,
+                    f"Polygon with {len(value)} points." if value is not None else None,
+                )
+            elif name not in ["confidence", "raw_value"]:
+                self.__setattr__(
+                    name,
+                    str(value) if value is not None else None,
+                )
+            self.page_id = page_id or item_page_id
+
+    def __str__(self, level=0) -> str:
```

Comment:
Don't change the signature of a magic method

Create another method that takes the `level` param, and in `__str__` call that method

Reply:
Fair. Though that makes different levels of recursion a bit harder to read.